From 6127cf8b026a870f436793d5c7a7b4e8c8e07d0a Mon Sep 17 00:00:00 2001
From: Anthony Adinolfi <ama39@cornell.edu>
Date: Mon, 27 Jun 2022 15:37:21 -0400
Subject: [PATCH] Experimental code edit to prevent the bug that mishandles
 "allowed HTML tags" field

(cherry picked from commit 79d78ecab32b5a1c0ba29f291e84245d9cce1d91)
---
 .../modules/filter/filter.filter_html.admin.js     | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/web/core/modules/filter/filter.filter_html.admin.js b/web/core/modules/filter/filter.filter_html.admin.js
index 54eed3ff6..d31500f9c 100644
--- a/web/core/modules/filter/filter.filter_html.admin.js
+++ b/web/core/modules/filter/filter.filter_html.admin.js
@@ -173,7 +173,17 @@
         }
 
         if (rule.restrictedTags.allowed.classes.length) {
-          setting += " class=\"".concat(rule.restrictedTags.allowed.classes.join(' '), "\"");
+
+          // Edit to protect the "allowed HTML tags" field from a recent bug that mishandles tags with all classes accepted (ama39 6/24/22)
+          // Issue on Drupal.org: https://www.drupal.org/project/drupal/issues/2710427
+          if (rule.restrictedTags.allowed.classes[0] == '' || rule.restrictedTags.allowed.classes[0] == '*') {
+            setting += " class";
+          }
+          else {
+            setting += " class=\"".concat(rule.restrictedTags.allowed.classes.join(' '), "\"");
+          }
+          // ---
+
         }
 
         setting += '>';
@@ -194,4 +204,4 @@
     html += '</p>';
     return html;
   };
-})(jQuery, Drupal, _, document);
\ No newline at end of file
+})(jQuery, Drupal, _, document);
